submariner/multiple-cable-drivers.md

```diff
@@ -0,0 +1,132 @@
+## Support fine-grained cluster connectivity and multiple cable drivers
+
+## Summary
+
+Submariner currently assumes full mesh by default and provision VPN connections between all participating clusters. For
+some use cases, it makes sense to offer finer-grained connectivity options. This could also improve overall
+scalability where  full mesh is not required.
+
+Submariner supports several cable driver implementations (libreswan, wireguard et al) however only one can be
+configured within the same cluster set, that is all clusters that join a broker must use the same cable driver.
+
+There are use cases that involve supporting a mix of cable drivers or cable driver configurations within the same
+cluster set. For example, a cluster set could contain three clusters, two that are on-premise and the other
+in the cloud. Connections to the cloud cluster will need to use encryption while the on-premise clusters can use
+unencrypted connections, as they’re connected by a private link, to avoid the performance overhead of encryption.
+This could be achieved using different cable driver implementations or the same cable driver implementation with
+differing encryption configurations.
+
+## Proposal
+
+This enhancement proposes a mechanism to allow users to specify their intent as to which clusters to connect and how to
+connect them. The default would still be a full mesh with an homogeneous cable driver.
+
+The standard mechanism in Kubernetes to specify intent is by creating a Kubernetes API resource whose effect is eventually
+realized into the cluster's desired state. To extend the Kubernetes API, one could either use the generic Kubernetes
+`ConfigMap` or use a custom resource definition (CRD). There's pros and cons to each but a CRD seems more suitable for
+this enhancement.
+
+This following CRD is proposed:
+
+```Go
+type ClusterConnectivityPolicies struct {
```

Comment:
Would it be better to have ClusterConnectivityPolicies and ClusterConnectivityPoliciesBinding? So that we can reuse policies. The former defining policies and the latter deciding which clusters it should apply to

Reply:
The idea is that there's only one `ClusterConnectivityPolicies` per cluster set so there would be no reuse.